File: frontend/src/types/task.ts
export type TaskStatus = '已发布' | '已认领' | '已完成';

export interface Task {
  id: string;
  title: string;
  content: string; // 详情描述
  date: string;    // 发布日期
  status: TaskStatus;
  completionTime?: string; // 完成时间
  estimatedCompletionTime?: string; // 预计完成时间
}

File: backend/src/main/java/com/task/management/entity/Task.java
package com.task.management.entity;

import lombok.Data;

import java.time.LocalDateTime;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;

@Data
public class Task {
    
    @TableId(type = IdType.AUTO)
    private Long id;

    private String title;

    private String content;

    private String status;

    //预计完成时间
    private LocalDateTime estimatedCompletionTime;

    private LocalDateTime createTime;

    private LocalDateTime updateTime;
    
    //实际完成时间
    private LocalDateTime completionTime;

}
